tests/modules/images/regenerate-existing-images/background-process/perflab-background-job-test.php

```diff
@@ -0,0 +1,148 @@
+		$this->assertEquals( 'perflab_job_attempts', constant( $job_class . '::JOB_ATTEMPTS_META_KEY' ) );
+		$this->assertEquals( 'perflab_job_lock', constant( $job_class . '::JOB_LOCK_META_KEY' ) );
+		$this->assertEquals( 'perflab_job_errors', constant( $job_class . '::JOB_ERRORS_META_KEY' ) );
+		$this->assertEquals( 'perflab_job_status', constant( $job_class . '::JOB_STATUS_META_KEY' ) );
+		$this->assertEquals( 'perflab_job_queued', constant( $job_class . '::JOB_STATUS_QUEUED' ) );
+		$this->assertEquals( 'perflab_job_running', constant( $job_class . '::JOB_STATUS_RUNNING' ) );
+		$this->assertEquals( 'perflab_job_partial', constant( $job_class . '::JOB_STATUS_PARTIAL' ) );
+		$this->assertEquals( 'perflab_job_complete', constant( $job_class . '::JOB_STATUS_COMPLETE' ) );
+		$this->assertEquals( 'perflab_job_failed', constant( $job_class . '::JOB_STATUS_FAILED' ) );
+	}
+
+	public function test_job_id_is_zero() {
+		$this->job = new Perflab_Background_Job();
+
+		$this->assertSame( 0, $this->job->job_id );
+	}
+
+	public function test_set_status_false_for_invalid_status() {
+		$this->job = new Perflab_Background_Job();
+
+		$result = $this->job->set_status( 'invalid_status' );
+		$this->assertFalse( $result );
+	}
+
+	/**
+	 * @covers ::create
+	 */
+	public function test_create_job() {
+		$job_data = array(
+			'post_id'          => 10,
+			'some_random_data' => 'some_random_string',
+		);
+
+		$job = Perflab_Background_Job::create( 'test_job', $job_data );
+
+		$this->assertInstanceOf( Perflab_Background_Job::class, $job );
+	}
+
+	/**
+	 * @covers ::set_error
+	 * @covers ::get_attempts
+	 */
+	public function test_set_error() {
+		$error      = new WP_Error();
+		$error_data = array(
+			'test_error_data' => 'descriptive_infomation',
+		);
+		$error->add_data( $error_data, 'perflab_job_failure' );
+		$this->job = new Perflab_Background_Job();
+		$job_data  = array(
+			'post_id'          => 10,
+			'some_random_data' => 'some_random_string',
+		);
+
+		$this->job = Perflab_Background_Job::create( 'random', $job_data );
+		$this->job->set_error( $error );
+
+		$error_metadata = get_term_meta( $this->job->job_id, 'perflab_job_errors', true );
+		$attempts       = $this->job->get_attempts();
+
+		$this->assertSame( $error_data, $error_metadata );
+		$this->assertEquals( 1, $attempts );
+	}
+
+	public function test_job_should_not_run_non_existing_job() {
+		$this->job = new Perflab_Background_Job();
+
+		$run = $this->job->should_run();
+		$this->assertFalse( $run );
+	}
+
+	public function test_job_should_not_run_for_completed_job() {
+		$this->job = new Perflab_Background_Job();
+		$this->job->create( 'test_job' );
+
+		// Mark job as complete.
+		$this->job->set_status( 'perflab_job_complete' );
+		$run = $this->job->should_run();
+
+		$this->assertFalse( $run );
+	}
+
+	/**
+	 * @covers ::lock
+	 * @covers ::get_status
+	 */
+	public function test_lock() {
```

Comment:
@ankitrox it looks like this test doesn't test the Job locking and instead tests the correct `queued` status is assigned on creation. Can we update this test to check the initial status and create another test to check the locking.